helpers/windows/pdh/pdh_query_windows.go

```diff
@@ -0,0 +1,377 @@
+		// A call is triggered again
+		if len(paths) == 1 && strings.Contains(paths[0], "*") && paths[0] == wildCardPath {
+			expdPaths, err = PdhExpandWildCardPath(utfPath)
+			if err == nil {
+				return paths, err
+			}
+		} else {
+			return paths, err
+		}
+	}
+
+	return nil, PdhErrno(syscall.ERROR_NOT_FOUND)
+}
+
+// Close closes the query and all of its counters.
+func (q *Query) Close() error {
+	return PdhCloseQuery(q.Handle)
+}
+
+// matchInstanceName will check first for instance and then for any objects names.
+func matchInstanceName(counterPath string) (string, error) {
+	matches := instanceNameRegexp.FindStringSubmatch(counterPath)
+	if len(matches) == 2 {
+		return returnLastInstance(matches[1]), nil
+	}
+	matches = objectNameRegexp.FindStringSubmatch(counterPath)
+	if len(matches) == 2 {
+		return matches[1], nil
+	}
+	return "", errors.New("query doesn't contain an instance name. In this case you have to define 'instance_name'")
+}
+
+// returnLastInstance will return the content from the last parentheses, this covers cases as `\WF (System.Workflow) 4.0.0.0(*)\Workflows Created`.
+func returnLastInstance(match string) string {
+	var openedParanth int
+	var innerMatch string
+	var matches []string
+	runeMatch := []rune(match)
+	for i := 0; i < len(runeMatch); i++ {
+		char := string(runeMatch[i])
+
+		// check if string ends between parentheses
+		if char == ")" {
+			openedParanth -= 1
+		}
+		if openedParanth > 0 {
+			innerMatch += char
+		}
+		if openedParanth == 0 && innerMatch != "" {
+			matches = append(matches, innerMatch)
+			innerMatch = ""
+		}
+		// check if string starts between parentheses
+		if char == "(" {
+			openedParanth += 1
+		}
+	}
+	if len(matches) > 0 {
+		return matches[len(matches)-1]
+	}
+	return match
+}
+
+// getCounterValue will retrieve the counter value based on the format applied in the config options
+func getCounterValue(counter *Counter) CounterValue {
+	counterValue := CounterValue{Instance: counter.instanceName, Err: CounterValueError{CStatus: 0}}
+	switch counter.format {
+	case PdhFmtLong:
+		_, value, err := PdhGetFormattedCounterValueLong(counter.handle)
+		if err != nil {
+			counterValue.Err.Error = err
+			if value != nil {
+				counterValue.Err.CStatus = value.CStatus
+			}
+		} else {
+			counterValue.Measurement = value.Value
+		}
+	case PdhFmtLarge:
+		_, value, err := PdhGetFormattedCounterValueLarge(counter.handle)
+		if err != nil {
+			counterValue.Err.Error = err
+			if value != nil {
+				counterValue.Err.CStatus = value.CStatus
+			}
+		} else {
+			counterValue.Measurement = value.Value
+		}
+	case PdhFmtDouble:
+		_, value, err := PdhGetFormattedCounterValueDouble(counter.handle)
+		if err != nil {
+			counterValue.Err.Error = err
+			if value != nil {
+				counterValue.Err.CStatus = value.CStatus
+			}
+		} else {
+			counterValue.Measurement = value.Value
+			fmt.Println(value.Value)
```

Comment:
This is added from the one in metricbeat.  maybe a leftover debug println?

Reply:
> This is added from the one in metricbeat. maybe a leftover debug println?

i think it's due to my mistake. I added this.
anyway, I'll remove it